src/static/js/chapter.js

```diff
@@ -256,6 +256,102 @@ function setDiscussionCount() {
   }
 }
 
+function indexHighlighter() {
+    //Only activate this if IntersectionObserver is supported
+  if('IntersectionObserver' in window){
+    var indexScroller = document.querySelector('.index-scroller');
+
+    // Check if user has set reduced motion and only continue if not
+    var hasOSReducedMotion = window.matchMedia('(prefers-reduced-motion: reduce)').matches;
+    if (hasOSReducedMotion) {
+      console.log('User has set prefers-reduced-motion to ' + hasOSReducedMotion + ' so not highlighting the current section in chapter index');
+    } else {
+
+      // Check if 'position:sticky' is supported (as this is not great UX when not so don't bother)
+      // Add the sticky class (which sets 'position:sticky') and then test if that stuck :-)
+      indexScroller && indexScroller.parentNode && indexScroller.parentNode.classList.add('sticky');
+      var indexScrollerStyles = getComputedStyle(indexScroller.parentNode);
+      if (indexScrollerStyles && indexScrollerStyles.position) {
+        //Handle vendor prefixes (still required for Safari 12)
+        if (indexScrollerStyles.position.endsWith('sticky')) {
+
+          // Restrict the page height of the index to the page-height, as we're going to scroll this.
+          indexScroller.classList.add('page-height');
+
+          var indexItems = {};
+          var currentItem;
+
+          // Create a function to handle highlighting a new index item
+          // that will be called y bthe IntersectionObserver
+          function highlightIndexEntry(link) {
+            
+            var oldItem = currentItem;

```

Comment:
A simpler way to add/remove the `active` class, without the bookkeeping:

```js
document.querySelector('.index-scroller .active').classList.remove('active');
var index = document.querySelector(`.index-scroller a[href="#${link}"]`).parentNode;
index.classList.add('active');
```

Reply:
I was trying to avoid the querySelector call for perf reasons. And you have two here. Think I'm over-worrying about that?

Reply:
I don't see any cause for concern with these selectors.